cpp/include/raft/core/mdarray.hpp

```diff
@@ -914,4 +914,34 @@ auto reshape(const array_interface_type& mda, extents<Extents...> new_shape)
   return reshape(mda.view(), new_shape);
 }
 
-}  // namespace raft
\ No newline at end of file
+/**
+ * \brief Turns linear index into coordinate.  Similar to numpy unravel_index.
+ *
+ * \code
+ *   auto m = make_host_matrix<float>(7, 6);
+ *   auto m_v = m.view();
+ *   auto coord = unravel_index(2, m.extents(), typename decltype(m)::layout_type{});
+ *   std::apply(m_v, coord) = 2;
+ * \endcode
+ *
+ * \param idx    The linear index.
+ * \param shape  The shape of the array to use.
+ * \param layout Must be `layout_c_contiguous` (row-major) in current implementation.
+ *
+ * \return A std::tuple that represents the coordinate.
+ */
+template <typename LayoutPolicy, std::size_t... Exts>
+MDSPAN_INLINE_FUNCTION auto unravel_index(size_t idx,
```

Comment:
`std::size_t idx` as even extents are expressed as that type?

Reply:
I haven't tested the updated mdspan and its new extent type yet.

Reply:
Fair. We'll update it when we update mdspan

Reply:
Could this parameter possibly be optional with a default type? For those wishing to use the template param directly?

Reply:
Done.

Reply:
Am I missing the updates?

Reply:
I changed the parameter type of `std::size_t idx` into template `Idx idx`